### PR TITLE
Do not download model again

### LINKED ISSUE #18: fix: Do not download models already on disk
Before downloading a models, checking if it is already on disk, avoid downloading it multiple times.

### PR DESCRIPTION
**Pull request by feloy**
Fixes #18

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
would be nice (as for many others PR) to add more unit tests

File: packages/backend/src/managers/applicationManager.spec.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  vi.spyOn(os, 'homedir').mockReturnValue('/home/user');
+  const manager = new ApplicationManager({} as GitManager, {} as RecipeStatusRegistry, {} as ExtensionContext);
+  expect(manager.appUserDirectory).toMatch(/^\/home\/user/);
+});
+
+test('getLocalModels should return models in local directory', () => {
+  vi.spyOn(os, 'homedir').mockReturnValue('/home/user');
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  vi.spyOn(fs, 'readdirSync').mockImplementation((dir: string): any => {
+    // TODO(feloy): fix any
```
Comment (feloy) [inline packages/backend/src/managers/applicationManager.spec.ts:20]: The difficulty with readdirSync is that it returns either Dirent[] or string[] depending on the options passed. I'm not sure how to handle this without using any here...

Comment (benoitf) [reply packages/backend/src/managers/applicationManager.spec.ts:49]:
```ts
const readdirSyncMock: MockInstance<[path: fs.PathLike], string[]> = vi.spyOn(fs, 'readdirSync') as unknown as MockInstance<[path: fs.PathLike], string[]>;
readdirSyncMock.mockImplementation((path: fs.PathLike) => {
```

Comment (benoitf) [reply packages/backend/src/managers/applicationManager.spec.ts:49]: (that said on unit test I'm fine to use `any` type)

Comment (feloy) [reply packages/backend/src/managers/applicationManager.spec.ts]: Thanks

**Pull request merged**